### ISSUE TITLE
delayed inventory update on MineRLObtainDiamond-v0

### ISSUE THREAD

**Issue by SaminYeasar** [minerllabs/minerl#471]
Hi, I think there's a delay in updating the inventory in `MineRLObtainDiamond-v0`. Here I have saved the experiences (reward, frame, inventory, etc.) during evaluation. At timestep l=429 reward is 1, while the `log` is zero, `log` appears to be 1 at timestep l=431. Similarly, I get reward 2 and `plank` appears next-timestep. While in Human data the inventory updates at an instant.
![Screenshot from 2021-04-02 04-17-52](https://user-images.githubusercontent.com/25709272/113405165-5f0cd580-9377-11eb-9dd9-11ed463e1e91.png)

**Comment by KarolisRam** [comment]
I can confirm the reward/inventory change mismatch. Attaching a notebook I used to reproduce it using a set seed and hardcoded actions.
[test reward vs inventory change by frame.ipynb.zip](https://github.com/minerllabs/minerl/files/6249037/test.reward.vs.inventory.change.by.frame.ipynb.zip)

**Comment by shwang** [comment]
This is true for all environments -- in fact we [explicitly account for this delay in BASALT tests](
https://github.com/minerllabs/minerl/blob/097b36b3db335ca6e2ea2ed44590d343ee27c732/tests/item_variants_integration_test.py#L159-L164) now.

@brandonhoughton told me the reason why we have this delay at one point. Could you repeat the reason here in case someone wants to work on this in the future?

Also it probably would be a good doc change for us to at least document or warn about this problem in the observation space docs: https://minerl.readthedocs.io/en/latest/environments/handlers.html#visual-observations-pov-third-person

**Comment by brandonhoughton** [comment]
Some handlers work on information from the server - so we do client tick, then server tick for a step. If you need the server to tell you something it won't be processed until the next tick. That being said I'd expect that you can claim things in your own inventory on pickup right away - likely this is a result of the reward handler being on the server so it sees the change before you, sends it via a custom message to the client. Then on the next step the change is visible to the client
